_posts/2022-08-05-hacking-for-cruise-control-ui.md

```diff
@@ -0,0 +1,241 @@
+---
+layout: post
+title: "Hacking Strimzi for Cruise Control UI"
+date: 2022-10-31
+author: kyle_liberti
+---
+
+For some time now, Strimzi has offered Kubernetes native support Cruise Control allowing users to safely configure and communicate with Cruise Control through Strimzi custom resources.
+This is great but sometimes it is a better user experience to interact with applications like Cruise Control through a UI.
+Although Strimzi does not offer direct support for the [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) it can still be run alongside and connected to a Strimzi managed Cruise Control instance.
+
+This post will walk through how to do so! 
+
+## Prerequisites
+
+For starters we will need a Strimzi managed Kafka cluster with Cruise Control enabled.  
+
+```
+                           *  *
+                        *        *
+                       *  Cruise  *
+                       *  Control *
+                        *        *
+                           *  *
+                            ^                   *  *         *  *          *  *
+                            |                *        *   *        *    *        *
++----------+                |               *  Kafka   * *  Kafka   *  *  Kafka   *
+|          |               *  *             *   Pod    * *   Pod    *  *   Pod    *
+|  Kafka   |            *        *           *        *   *        *    *        *
+| Resource | <------+  * Cluster  * ----->      *  *         *  *          *  *
+|          | +------>  * Operator *
+|          |            *        *              *  *         *  *          *  *
++----------+               *  *              *        *   *        *    *        *
+                                            * Zookeeper* * Zookeeper*  * Zookeeper*
+                                            *   Pod    * *   Pod    *  *   Pod    *
+                                             *        *   *        *    *        *
+                                                *  *         *  *          *  *
+
+Kubernetes Land                                          Kafka Cluster
+```
+Since the [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) does not support HTTPs and Strimzi does not support creating custom Cruise Control API user roles we will need to disable Cruise Control SSL and HTTP basic authentication settings. 
+
+We can do so by configuring a Kafka custom resource like this:
+
+``` 
+# kafka-cruise-control.yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  cruiseControl:
+    config:
+      webserver.security.enable: false
+      webserver.ssl.enable: false  
+```
+
+## Cruise Control UI
+
+With our Strimzi managed Kafka cluster deployed, we can now create a Cruise Control UI instance to run alongside it.
```

Comment:
You took the proxy approach but have you considered building a new Cruise Control image with the UI added to it?

Reply:
Added new method for building a custom Cruise Control image